src/sharing/SharingScreen.js

```diff
@@ -0,0 +1,498 @@
+              <Image
+                source={{ uri: sharedData.sharedImageUrl }}
+                width={200}
+                height={200}
+                style={styles.imagePreview}
+              />
+            )}
+            <AnimatedScaleComponent visible={isStreamFocused}>
+              <StreamAutocomplete filter={stream} onAutocomplete={this.handleStreamAutoComplete} />
+            </AnimatedScaleComponent>
+            <Input
+              placeholder="Stream"
+              value={stream}
+              onChangeText={this.handleStreamChange}
+              onFocus={this.focusStream}
+              onChange={this.focusStream}
+              onBlur={this.blurStream}
+              selectTextOnFocus
+            />
+            <View />
+            <AnimatedScaleComponent visible={isTopicFocused}>
+              <TopicAutocomplete
+                text={topic}
+                onAutocomplete={this.handleTopicAutoComplete}
+                isFocused={isTopicFocused}
+                narrow={narrow}
+              />
+            </AnimatedScaleComponent>
+            <Input
+              placeholder="Topic"
+              value={topic}
+              onFocus={this.focusTopic}
+              onBlur={this.blurTopic}
+              onChangeText={this.handleTopicChange}
+              editable={stream !== ''}
+            />
+            <Input
+              placeholder="Message"
+              value={message}
+              onChangeText={this.handleMessageChange}
+              multiline
+            />
+          </View>
+        </ScrollView>
+        <View style={styles.actions}>
+          <ZulipButton onPress={this.finishShare} style={styles.button} secondary text="Cancel" />
+          <ZulipButton
+            style={styles.button}
+            onPress={this.handleSend}
+            text="Send"
+            progress={sending}
+            disabled={!this.isSendButtonEnabled()}
+          />
+        </View>
+      </>
+    );
+  }
+}
+
+type ShareToPmProps = $ReadOnly<{|
+  ...$Exact<NavigationNavigatorProps<{||}, {| params: {| sharedData: SharedData |} |}>>,
+  dispatch: Dispatch,
+  auth: Auth,
+|}>;
+
+type ShareToPmState = $ReadOnly<{|
+  users: User[],
+  message: string,
+  choosingRecipients: boolean,
+  sending: boolean,
+|}>;
+
+class ShareToPmComponent extends React.Component<ShareToPmProps, ShareToPmState> {
+  constructor(props) {
+    super(props);
+    const { sharedData } = this.props.navigation.state.params;
+    this.state = {
+      users: [],
+      message: sharedData.type === 'text' ? sharedData.sharedText : '',
+      choosingRecipients: false,
+      sending: false,
+    };
+  }
+
+  setSending = () => {
+    this.setState({ sending: true });
+  };
+
+  toggleChoosingRecipients = () => {
+    this.setState(state => ({ choosingRecipients: !state.choosingRecipients }));
+  };
+
+  handleChooseRecipients = (users: Array<User>) => {
+    this.setState({ users });
+    this.toggleChoosingRecipients();
```

Comment:
I agree, it can be confusing. Removed the toggle function.

>There's also the fact that the "Choose recipients" button doesn't really work as a toggle switch between the two states, but rather to just open the modal (i.e., change the state from false to true); it gets hidden behind the modal, and therefore effectively disabled, when the modal opens. It might actually be good to reinforce this pattern by using ZulipButton's disabled prop.

The button does not get hidden behind, it's not rendered at all when `choosingRecipients` is true.

```js
    if (choosingRecipients) {
      return (
        <Modal>
          <ChooseRecipientsScreen onComplete={this.handleChooseRecipients} />
        </Modal>
      );
    }
```

So I think it's okay to not use the `disabled` prop.